packages/psss-server/src/routes/reports/SaveWeeklyReportRoute.ts

```diff
@@ -5,36 +5,37 @@
 
 import { RespondingError } from '@tupaia/utils';
 import { Route } from '../Route';
+import { WEEKLY_SURVEY_COUNTRY, WEEKLY_SURVEY_SITE } from '../../constants';
 import { validateIsNumber } from '../../utils';
 
-const SURVEY_CODE = 'PSSS_WNR';
-
 type WeeklyReportAnswers = {
-  PSSS_Sites: number;
-  PSSS_Sites_Reported: number;
   PSSS_AFR_Cases: number;
   PSSS_DIA_Cases: number;
   PSSS_ILI_Cases: number;
   PSSS_PF_Cases: number;
   PSSS_DLI_Cases: number;
+  PSSS_Sites?: number;
+  PSSS_Sites_Reported?: number;
 };
 
-export class SaveCountryWeeklyReportRoute extends Route {
+export class SaveWeeklyReportRoute extends Route {
   async buildResponse() {
     const { week } = this.req.query;
-    const { organisationUnitCode } = this.req.params;
-    const answers = mapReqBodyToAnswers(this.req.body);
+    const { countryCode, siteCode } = this.req.params;
+
+    const isSiteSurvey = !!siteCode;
+    const answers = mapReqBodyToAnswers(this.req.body, isSiteSurvey);
 
     return this.meditrakConnection?.updateOrCreateSurveyResponse(
-      SURVEY_CODE,
-      organisationUnitCode,
+      isSiteSurvey ? WEEKLY_SURVEY_SITE : WEEKLY_SURVEY_COUNTRY,
+      isSiteSurvey ? siteCode : countryCode,
```

Comment:
Might be nice to de-duplicate this whole 'siteCode'/'siteSurvey' logic in `DeleteWeeklyReportRoute.ts` but I can't think of a clean and easy way to do it off the top of my head so I'm fine with it 👍

Reply:
I could definitely use a helper for that, but I let this [WET](https://overreacted.io/the-wet-codebase/) on purpose. I thought this small duplication will actually make the code more maintainable than abstracting it.

In the (not too distant past) I believe I'd try to de-duplicate this, but recently I've been trying to "mature" my approach on DRY. The following articles have influenced me:
* https://overreacted.io/goodbye-clean-code/
* https://sandimetz.com/blog/2016/1/20/the-wrong-abstraction

Of course, the above resources do not "prove" that my decision is correct, but just some food for thought 😄

Reply:
This is really interesting @kael89 - I watched a similar presentation recently about avoiding hasty abstractions which was a light bulb moment for me. Check it out [here](https://kentcdodds.com/blog/aha-programming) if you are interested. This would make an interesting dev chat topic.

Reply:
..... quietly puts into dev chat agenda* 👀   @tcaiger / @kael89